velox/docs/develop/types.rst

```diff
@@ -145,3 +145,38 @@ Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
 (or [-69387-04-22T03:45:14.752, 73326-09-11T20:14:45.247]). The low 12 bits
 store timezone ID. Supported range of timezone ID is [1, 1680].
 The definition of timezone IDs can be found in ``TimeZoneDatabase.cpp``.
+
+Spark Types
+~~~~~~~~~~~~
+The data types in Spark have some semantic differences compared to those in 
+Presto. These differences require us to implement the same functions 
+separately for each system in Velox, such as min, max and collect_set. The 
+key differences are listed below.
+
+* Spark operates on timestamps with "microsecond" precision while Presto with 
+  "millisecond" precision.
+  Example::
+
+      SELECT min(ts)
+      FROM (
+          VALUES
+              (cast('2014-03-08 09:00:00.123456789' as timestamp)),
+              (cast('2014-03-08 09:00:00.012345678' as timestamp))
+      ) AS t(ts);
+      -- 2014-03-08 09:00:00.123456
+
+* In function comparisons, nested null values are handled as values.
```

Comment:
Does this apply apply to join, group by, order by keys? Does Spark allow these operations with array / row types and handle nested NULLs as values? Is this currently broken in Velox?

CC: @rui-mo @bikramSingh91